curvefs/src/client/rpcclient/metacache.cpp

```diff
@@ -0,0 +1,347 @@
+/*
+ *  Copyright (c) 2021 NetEase Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/*
+ * Project: curve
+ * Created Date: Thur Sept 4 2021
+ * Author: lixiaocui
+ */
+
+#include <iterator>
+#include <vector>
+#include "curvefs/src/client/rpcclient/metacache.h"
+#include "src/client/metacache.h"
+
+using ::curve::common::ReadLockGuard;
+using ::curve::common::WriteLockGuard;
+
+namespace curvefs {
+namespace client {
+namespace rpcclient {
+
+unsigned int seed = 123;
+
+bool MetaCache::GetTarget(uint32_t fsID, uint64_t inodeID,
+                          MetaserverID *metaServerID, butil::EndPoint *target,
+                          LogicPoolID *poolID, CopysetID *copysetID,
+                          uint32_t *patitionID, uint64_t *applyIndex,
+                          bool refresh) {
```

Comment:
it better to define a type `PartitionId` and encapsulates LogicalPoolId CopystId and PartitionId into one struct

Reply:
fix